### PR TITLE
feat(api): add hoc wrappers

### PR DESCRIPTION
**Pull request by arahansen**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,7 +113,7 @@
   "bundlesize": [
     {
       "path": "./dist/reflex.min.js",
-      "maxSize": "6.0 Kb"
```
Comment (obartra) [inline package.json]: Maybe 6.5?

File: src/layout.hoc.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
+// @flow
```
Comment (obartra) [inline src/layout.hoc.js]: Is a layout hoc needed?

Comment (arahansen) [inline src/layout.hoc.js]: I figured it'd be nice to have. But I don't really have a use case for it. So I'm fine dumping it

File: package.json (hunk 1 above)
Comment (arahansen) [inline package.json]: I was going to, but readme looks like its already set to 7.0kb which is why I didn't change it. Also why I went with 7k for the maxSize

**Comment by arahansen** [comment]
@obartra backed out layout changes.

File: package.json (hunk 1 above)
Comment (obartra) [inline package.json]: it's +1 for the css size

Comment (arahansen) [inline package.json]: gotcha

**Pull request merged**